client.go

```diff
@@ -281,7 +281,7 @@ func (c *Client) Pull(ctx context.Context, ref string, opts ...RemoteOpt) (Image
 	}
 	if pullCtx.Unpack {
 		if err := img.Unpack(ctx, pullCtx.Snapshotter); err != nil {
-			return nil, err
+			return nil, errors.Wrap(err, "failed to unpack image")
```

Comment:
nit: `errors.Wrapf(err, "failed to unpack image on snapshotter %s", pullCtx.Snapshotter)`?